popde/density_estimate.py

```diff
@@ -49,25 +57,65 @@ def __init__(self, data, backend='scipy', bandwidth=1., dim_names=None):
         """
         if len(data.shape) != 2:
             raise ValueError("Data must have shape (n_samples, n_features).")
+        self.ndim = self.data.shape[1]
 
         self.data = np.asarray(data)
```

Comment:
should this line go before line 58 (if len(data.shape) != 2:)? To transform data into an array before anything else, because lists for example don't have a list.shape method

Reply:
mmm, so we shouldn't be using a list as input in any case, if so there will be an error even earlier when doing data.shape, & I think it is 'correct' that we get an error.